ui/components/src/TimeSeriesTooltip/nearby-series.ts

```diff
@@ -68,8 +68,13 @@ export function checkforNearbyTimeSeries(
 
   const yValueCounts: Map<number, number> = new Map();
 
-  let closestTimestamp = null;
-  let closestDistance = Infinity;
+  // Only need to loop through first dataset source since getCommonTimeScale ensures xAxis timestamps are consistent
+  const firstTimeSeriesValues = data[0]?.values;
+  const closestTimestamp = getClosestTimestamp(firstTimeSeriesValues, cursorX);
```

Comment:
More context about this change is in PR #1330 (which I just merged into this branch)